### PR TITLE
Archive helper

### PR DESCRIPTION
**Pull request by morfi11**

### CODE REVIEW

**Comment by morfi11** [comment]
could not figure out how to remove the archived helpers from the search dialog. tries to fiddle with MapHelpers but did not work :(
need your help @noam-honig

File: src/app/asign-family/asign-family.component.ts [hunk 1]
```diff
@@ -77,7 +77,7 @@ export class AsignFamilyComponent implements OnInit, OnDestroy {
 
         if (this.isValidPhone()) {
             this.phone = cleanPhone;
-            let helper = await this.context.for(Helpers).findFirst(h => h.phone.isEqualTo(this.phone));
+            let helper = await this.context.for(Helpers).findFirst(h => h.phone.isEqualTo(this.phone).and(h.active()));
```
Comment (noam-honig) [inline src/app/asign-family/asign-family.component.ts:80]: Why? If the helper is inactive - but is coming to volunteer - we should be able to reactive them.
The way it is now will cause a duplicate phone number on login

File: src/app/families/FamilyDeliveries.ts [hunk 2]
```diff
@@ -83,6 +83,9 @@ export class FamilyDeliveries extends IdEntity {
         allowApiUpdate: Roles.admin
     });
     quantity = new QuantityColumn(this.context, { caption: getLang(this.context).quantity, allowApiUpdate: Roles.admin, dataControlSettings: () => ({ width: '100', inputType: 'number' }) });
+    isLargeQuantity() {
```
Comment (noam-honig) [inline src/app/families/FamilyDeliveries.ts:86]: should restrict this to you.
The large quantity doesn't always make sense (I have a team that uses the quantity for the cash amount they should charge the recipient)

File: src/app/select-helper/select-helper.component.ts [hunk 3]
```diff
@@ -2,7 +2,7 @@ import { Component, OnInit } from '@angular/core';
 
 import { MatDialogRef } from '@angular/material/dialog';
 import { Helpers, HelpersBase } from '../helpers/helpers';
-import { Context, FindOptions, ServerFunction, DialogConfig, SqlDatabase } from '@remult/core';
+import { Context, FindOptions, ServerFunction, DialogConfig, SqlDatabase, Filter } from '@remult/core';
```
Comment (noam-honig) [inline src/app/select-helper/select-helper.component.ts:5]: why?

File: src/app/select-helper/select-helper.component.ts [hunk 4]
```diff
@@ -237,6 +237,7 @@ function mapHelpers<hType extends HelpersBase>(helpers: hType[], getFamilies: (h
     helperId: h.id.value,
     name: h.name.value,
     phone: h.phone.displayValue,
+    isActive: h.active(),
```
Comment (noam-honig) [inline src/app/select-helper/select-helper.component.ts:240]: not sure it's needed here

File: src/app/families/FamilyDeliveries.ts (hunk 2 above)
Comment (morfi11) [reply src/app/families/FamilyDeliveries.ts:86]: I will restrict it to MLT. The version I based on did not have the new isSystemForMLT() - I will update it now

File: src/app/asign-family/asign-family.component.ts (hunk 1 above)
Comment (morfi11) [reply src/app/asign-family/asign-family.component.ts:80]: I am trying to remove them form the search but could not find how. 
currently was only able to remove in helpers-component so they do not appear in the list. 
the logic for the search is too complex for my feeble mind :) - can you show me?

**Pull request closed without merging**